arrows/ffmpeg/tests/decode_barcode.h

```diff
@@ -0,0 +1,76 @@
+/*ckwg +29
+ * Copyright 2018 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef FFMPEG_TEST_BARCODE_DECODE_H
+#define FFMPEG_CORE_TEST_BARCODE_DECODE_H
+
+#include <vital/types/image_container.h>
+
+// Ignore 8 pixels on either side of the barcode
+static int bc_buffer = 8;
+// Barcode lines two pixels wide and 4 pixels high
+static int bc_width = 2;
+static int bc_height = 4;
+static int bit_depth = 256;
+static int bc_area = bc_width*bc_height;
+
+// Decode barcodes from test frame images
+uint32_t decode_barcode(kwiver::vital::image_container_sptr img_ct)
```

Comment:
We should probably share a single copy of this code, and possibly the test data, between this and the VXL tests.  Note the VXL barcode reader is in the master branch now, you may need to merge in master to see it.  We could put the common code in core/tests for now, but really we should create a better place for common tests, data, and utility functions.